marklogic-data-hub-central/ui/src/components/entities/matching/ruleset-single-modal/ruleset-single-modal.tsx

```diff
@@ -76,40 +100,78 @@ const MatchRulesetModal: React.FC<Props> = (props) => {
       case 'exact':
       case 'reduce':
       case 'zip':
-        let propertyName = selectedProperty || '';
-
-        let matchRule: MatchRule = {
-          entityPropertyPath: propertyName,
-          matchType: matchType,
-          options: {}
-        }
+        {
+          let propertyName = selectedProperty || '';
 
-        let matchRuleset: MatchRuleset = {
-          name: propertyName,
-          weight: 0,
-          matchRules: [matchRule]
+          let matchRule: MatchRule = {
+            entityPropertyPath: propertyName,
+            matchType: matchType,
+            options: {}
+          }
+  
+          let matchRuleset: MatchRuleset = {
+            name: propertyName,
+            weight: 0,
+            matchRules: [matchRule]
+          }
+  
+          //check for duplicate
+          let propertyRulesets = curationOptions.activeStep.stepDefinition['matchRulesets'].filter(matchRuleset => matchRuleset['name'] === propertyName)
```

Comment:
Does "stepDefinition" here refer to the configuration of this step? If so, I highly recommend not using the term "step definition", which in DHF means "an artifact that identifies a name and type and module path for a step". I actually wouldn't expect anything in between "activeStep" and "matchRulesets" either, since matchRulesets is a top-level property of a step. If "activeStep" is actually a wrapper containing the step config and other properties, I would call it something besides a "step", because its name to me implies that matchRulesets should be a property of it.

Reply:
stepDefinition = stepArtifact from payload. I can rename it to that.